js/lib/beautify.js

```diff
@@ -1115,11 +1150,87 @@
                 flags.in_case = false;
                 return;
             }
+            
+            var space_before = true;
+            var space_after = true;
+            var in_ternary = false;
+                
+            if (current_token.text === ':') {
+                if (flags.ternary_depth === 0) {
+                    // Colon is invalid javascript outside of ternary and object, but do our best to guess what was meant.
+                    space_before = false;
+                } else {
+                    flags.ternary_depth -= 1;
+                    in_ternary = true;
+                }
+            } else if (current_token.text === '?') {
+                flags.ternary_depth += 1;
+            }
+            
+            // let's handle the operator_position option prior to any conflicting logic
+            if (typeof opt.operator_position !== 'undefined' && in_array(current_token.text, OP_POSITION_OPERATORS)) {
```

Comment:
Handling the defaulting in the opt setting above means you don't have to check for undefined here.